scripts/azure/storage-manager.sh

```diff
@@ -0,0 +1,169 @@
+    exit 1
+fi
+
+[ "$objectstorage_type_sc" = "azure" ] && buckets_sc=$(echo "${sc_config}" | yq4 '.objectStorage.buckets.*' -)
+[ "$objectstorage_type_wc" = "azure" ] && buckets_wc=$(echo "${wc_config}" | yq4 '.objectStorage.buckets.*' -)
+
+RESOURCE_GROUP=$(echo "${common_config}" | yq4 '.objectStorage.azure.resourceGroup')
+STORAGE_ACCOUNT=$(echo "${common_config}" | yq4 '.objectStorage.azure.storageAccountName')
+CONTAINERS=$( { echo "${buckets_sc:-}"; echo "${buckets_wc:-}"; } | sort | uniq | tr '\n' ' ' | sed s'/.$//')
+
+log_info "Operating on containers: ${CONTAINERS// /', '}"
+
+function usage() {
+    echo "Usage:" 1>&2
+    echo " $0 create" 1>&2
+    echo " $0 delete" 1>&2
+}
+
+if [ "${#}" -lt 1 ]; then
+  usage
+  exit 1
+fi
+
+case "$1" in
+create)
+    ACTION=$CREATE_ACTION
+    ;;
+delete)
+    ACTION=$DELETE_ACTION
+    ;;
+esac
+shift
+
+function create_resource_group() {
+
+    log_info "checking if resource group exists" >&2
+    GROUP_EXISTS=$(az group list --query '[].name' | jq --arg group "${RESOURCE_GROUP}" '. | index($group)')
+    if [ "$GROUP_EXISTS" != null ]; then
+        log_info "resource group [${RESOURCE_GROUP}] already exists" >&2
+        log_info "continue using this group ? (y/n)" >&2
+        read -r -n 1 cmdinput
+        case "$cmdinput" in
+        y|Y) return ;;
+        *) exit 0 ;;
+        esac
+    else
+        log_info "resource group [${RESOURCE_GROUP}] does not exist, creating it now" >&2
+        az group create \
+            --name "${RESOURCE_GROUP}" \
+            --location "${AZURE_LOCATION}" --only-show-errors
+    fi
+}
+
+function create_storage_account() {
+
+    log_info "checking storage account availability" >&2
+    out=$(az storage account check-name --only-show-errors --name "${STORAGE_ACCOUNT}")
+    ACCOUNT_AVAILABLE=$(echo "$out" | jq -r .nameAvailable)
+    REASON=$(echo "$out" | jq -r .reason)
+    case $ACCOUNT_AVAILABLE in
+        false)
+            if [ "$REASON" == "AccountNameInvalid" ]; then
+                log_info "Account name invalid, must be only contain numbers and lowercase letters"
+                exit 0
+            elif [ "$REASON" == "AlreadyExists" ]; then
+                log_info "storage account [${STORAGE_ACCOUNT}] already exists" >&2
+                log_info "contnue using this account ? (y/n)" >&2
+                read -r -n 1 cmdinput
+                case "$cmdinput" in
+                y) return ;;
+                n) exit 0 ;;
+                esac
+            fi
+            ;;
+        true)
+            log_info "creating storage account ${STORAGE_ACCOUNT}"
+            az storage account create \
+            --name "$STORAGE_ACCOUNT" \
+            --resource-group "${RESOURCE_GROUP}" \
+            --location swedencentral \
+            --sku Standard_RAGRS \
+            --kind StorageV2 \
+            --allow-blob-public-access false --only-show-errors
+            ;;
+    esac
```

Comment:
`az storage account create` is idempotent, no need to check if it already exists.

```suggestion
        log_info "creating storage account ${STORAGE_ACCOUNT}"
        az storage account create \
            --name "${STORAGE_ACCOUNT}" \
            --resource-group "${RESOURCE_GROUP}" \
            --location ${AZURE_LOCATION} \
            --sku Standard_RAGRS \
            --kind StorageV2 \
            --allow-blob-public-access false --only-show-errors
```

Reply:
same as resource group.